pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -1167,8 +1167,7 @@ public ServiceUnitId getServiceUnitId(TopicName topicName) throws Exception {
                                 pulsar.getBrokerService().getMultiLayerTopicMap().get(namespaceName.toString())
                                         .forEach((__, bundle) -> {
                                             bundle.forEach((topicName, topic) -> {
-                                                if (topic instanceof NonPersistentTopic
-                                                        && ((NonPersistentTopic) topic).isActive()) {
```

Comment:
Have you checked why do we added `isActive` check in the first place?
Will it affect other use cases if we remove this?